qemu/tests/qemu_guest_agent.py

```diff
@@ -1828,6 +1828,27 @@ def gagent_check_nonexistent_cmd(self, test, params, env):
         else:
             test.fail("Should return error info.")
 
+    @error_context.context_aware
+    def gagent_check_with_migrate(self, test, params, env):
+        """
+        Migration test with guest agent service running.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        error_context.context("Migrate guest while guest agent service is"
+                              " running.", logging.info)
+        self.vm.monitor.migrate_set_speed(params.get("mig_speed", "1G"))
+        self.vm.migrate()
+        error_context.context("Recreate a QemuAgent object after vm"
+                              " migration.", logging.info)
+        self.gagent = False
+        args = [params.get("gagent_serial_type"), params.get("gagent_name")]
+        self.gagent_create(params, self.vm, *args)
```

Comment:
```
 188     def gagent_create(self, params, vm, *args):                                 
 189         if self.gagent:                                                         
 190             return self.gagent  
```
I did not get where to remove `self.gagent` in the script, so it should be the old gagent object i guess.
However, this case need restart the socket after migration finished,  restart socket will be implemented during `QemuAgent.init`.   So i think the current design did not restart the socket, please double check it!

Reply:
@yanan-fu 
The following code of gagent_create() function will create a new socket.
gagent = guest_agent.QemuAgent(vm, gagent_name, gagent_serial_type,
                                       filename, get_supported_cmds=True)
it will create a new socket connected to guest agent socket, and I set self.gagent = False after migration.
So it works.
thanks.

Reply:
@xiagao i think you did not get the point, let me be more clear.
Yes,  `gagent = guest_agent.QemuAgent` will recreate the socket, however you didn't call it directly, you use `self.gagent_create`.
In `gagent_create`, it will check if there is one gagent object first, and recreate new object only if there is none.  Please check the previous comment.

Reply:
I my previous comment,  "I set self.gagent = False after migration.

> @xiagao i think you did not get the point, let me be more clear.
> Yes, `gagent = guest_agent.QemuAgent` will recreate the socket, however you didn't call it directly, you use `self.gagent_create`.

Before I use self.gagent_create(), set self.gagent = False already..... Do you see it?
Or I miss your point?

> In `gagent_create`, it will check if there is one gagent object first, and recreate new object only if there is none. Please check the previous comment.

Reply:
@yanan-fu

Reply:
@xiagao  ah~  You are right, i miss it in the previous comment!